org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/notifications/NotificationsTracker.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2017 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.core.internal.notifications;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+
+public class NotificationsTracker {
+
+  // visible for testing
+  public static final String FILENAME = "lastEventPolling.data";
+
+  private ZonedDateTime lastEventPolling;
+
+  private final Path lastEventPollingPath;
+
+  public NotificationsTracker(Path basedir) {
+    if (!basedir.toFile().isDirectory()) {
+      try {
+        Files.createDirectories(basedir);
+      } catch (IOException e) {
+        // ignore
+      }
+    }
+    lastEventPollingPath = basedir.resolve(FILENAME);
+  }
+
+  // visible for testing
+  public synchronized ZonedDateTime getLastEventPolling() {
+    if (lastEventPolling == null) {
+      lastEventPolling = readFromFile();
+      if (lastEventPolling == null) {
+        lastEventPolling = ZonedDateTime.now();
+      }
+    }
+    return lastEventPolling;
+  }
+
+  // visible for testing
+  public synchronized void setLastEventPolling(ZonedDateTime time) {
+    lastEventPolling = time;
+    writeToFile(time);
+  }
+
+  private ZonedDateTime readFromFile() {
+    if (!lastEventPollingPath.toFile().isFile()) {
+      return null;
+    }
+    try {
+      long millis = Long.parseLong(new String(Files.readAllBytes(lastEventPollingPath)));
+      return ZonedDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneOffset.systemDefault());
+    } catch (IOException | NumberFormatException e) {
+      // ignore
+    }
+    return null;
+  }
+
+  private void writeToFile(ZonedDateTime time) {
+    try {
+      Files.write(lastEventPollingPath, Long.toString(time.toInstant().toEpochMilli()).getBytes());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "getBytes" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1943)